### PR TITLE
Fixed for empty alt value in react-a11y-img-has-alt rule.

### PR DESCRIPTION
**Pull request by t-ligu**
For img element,it must has alt attribute, empty alt must has role='presentation' and non-empty alt must has no role='presentation'.

Fixed a bug, for unit test, if using file data, the file name must has suffix '.tsx'.

### CODE REVIEW

File: src/reactA11yImgHasAltRule.ts [hunk 1]
```diff
@@ -10,17 +10,24 @@ import {
...
+  return `<${tagName}> elements must have an non-empty alt attribute or \
+use empty alt attribute and role='presentation' for presentational images. \
 A reference for the presentation role can be found at https://www.w3.org/TR/wai-aria/roles#presentation.`;
 }
 
-export function getFailureStringEmptyAlt(tagName: string): string {
-  return `The value of 'alt' attribute in <${tagName}> tag is undefined or empty. \
-Add more details in 'alt' attribute or use role='presentation' for presentational images. \
+export function getFailureStringEmptyAltAndNotPresentationRole(tagName: string): string {
+  return `The value of alt attribute in <${tagName}> tag is empty and role value is not presentation. \
```
Comment (ipip2005) [inline src/reactA11yImgHasAltRule.ts]: Maybe `or`?

File: src/reactA11yImgHasAltRule.ts [hunk 2]
```diff
@@ -10,17 +10,24 @@ import {
...
+Add more details in alt attribute or use empty alt attribute and role='presentation' for presentational images. \
```
Comment (ipip2005) [inline src/reactA11yImgHasAltRule.ts]: `Add more details in alt attribute or use empty alt attribute and role='presentation' for presentational images.` better be
`Specify role to equal 'presentation' when 'alt' attribute is empty.`

File: src/reactA11yImgHasAltRule.ts [hunk 3]
```diff
@@ -10,17 +10,24 @@ import {
...
+A reference for the presentation role can be found at https://www.w3.org/TR/wai-aria/roles#presentation.`;
+}
+
+export function getFailureStringNonEmptyAltAndPresentationRole(tagName: string): string {
+  return `The value of alt attribute in <${tagName}> tag is non-empty and role value is presentation. \
+Remove role='presentation' or use empty alt attribute and role='presentation' for presentational images. \
```
Comment (ipip2005) [inline src/reactA11yImgHasAltRule.ts]: `Remove role='presentation' or use empty alt attribute and role='presentation' for presentational images.`
better be
`Specify 'alt' attributeto be empty when role attributes equals 'presentation'.

File: src/reactA11yImgHasAltRule.ts [hunk 4]
```diff
@@ -31,7 +38,8 @@ export class Rule extends Lint.Rules.AbstractRule {
   public static metadata: ExtendedMetadata = {
     ruleName: 'react-a11y-img-has-alt',
     type: 'maintainability',
-    description: 'Enforce that an `img` element contains the `alt` attribute or `role="presentation"` for decorative image.',
+    description: 'Enforce that an `img` element contains the non-empty `alt` attribute or' +
```
Comment (ipip2005) [inline src/reactA11yImgHasAltRule.ts]: This description is confusing, A or B and C is not easy to understand. Better be:
'Enforce that an `img` element contains the non-empty `alt` attribute. For decorative images, using empty alt attribute and `role="presentation"`.`

File: src/reactA11yImgHasAltRule.ts (hunk 2 above)
Comment (t-ligu) [inline src/reactA11yImgHasAltRule.ts]: Fixed, thanks.

File: src/reactA11yImgHasAltRule.ts (hunk 3 above)
Comment (t-ligu) [inline src/reactA11yImgHasAltRule.ts]: Fixed, thanks.

File: src/reactA11yImgHasAltRule.ts (hunk 4 above)
Comment (t-ligu) [inline src/reactA11yImgHasAltRule.ts]: Fixed, thanks.

**Pull request closed without merging**